djstripe/models.py

```diff
@@ -357,35 +368,92 @@ def can_charge(self):
             self.date_purged is None
 
     def has_active_subscription(self):
```

Comment:
Customer.has_active_subscription() seems to only be called via utils.subscriber_has_active_subscription(subscriber) which also checks for staff status. One approach might be to leave this as is, and then have a similar set of functionality based on plans, maybe something like utils.subscriber_has_active_plan(subscriber, plan_id).